src/mlpack/tests/main_tests/logistic_regression_test.cpp

```diff
@@ -0,0 +1,378 @@
+/**
+  * @file logistic_regression_test.cpp
+  * @author B Kartheek Reddy
+  *
+  * Test mlpackMain() of logistic_regression_main.cpp.
+ **/
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+static const std::string testName = "LogisticRegression";
+
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/logistic_regression/logistic_regression_main.cpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+
+#include <boost/test/unit_test.hpp>
+
+using namespace mlpack;
+
+
+struct LogisticRegressionTestFixture
+{
+ public:
+  LogisticRegressionTestFixture()
+  {
+  	try { 
+    	// Cache in the options for this program.
+    	CLI::RestoreSettings(testName);
+    } catch (std::invalid_argument e) {
+    	Log::Fatal << "Invalid Test Name : " << e.what() << std::endl;
```

Comment:
I'm not sure this is necessary; we haven't had a problem with it before.  Were you having issues that required this code?

Reply:
No. I didn't have any issues with the code. I had just added it for safety, if in any case it produces an exception.

Reply:
Should I remove it?

Reply:
Yes, I think removing it is fine.  The method itself should throw an exception if there is an issue.